packages/node-opcua-server/test/test_server_issue_1303.ts

```diff
@@ -0,0 +1,87 @@
+import net from "net";
+import os from "os";
+import should from "should";
+import { OPCUAServer } from "..";
+
+async function findAvailablePort(): Promise<number> {
+    return new Promise((resolve) => {
+        const server = net.createServer();
+        server.listen(0);
+        server.on("listening", function () {
+            const port = (server.address() as net.AddressInfo).port;
+            console.log(`INFO: Found available port ${port}`);
+            server.close(() => resolve(port));
+        });
+    });
+}
+
+function findLanIp(): string | undefined {
+    const networkInterfaces = os.networkInterfaces();
+
+    for (const iface of Object.values(networkInterfaces)) {
+        if (!iface) {
+            continue;
+        }
+
+        for (const alias of iface) {
+            if (alias.family === "IPv4" && !alias.internal) {
+                return alias.address;
+            }
+        }
+    }
+
+    return undefined;
+}
+
+function checkServer(host: string | undefined, port: number): Promise<boolean> {
+    return new Promise((resolve) => {
+        const socket = net.createConnection(port, host, () => {
+            socket.end();
+            resolve(true);
+        });
+
+        socket.on("error", () => {
+            resolve(false);
+        });
+    });
+}
+
+async function testServerStartupAndShutdown(host: string | undefined, port: number) {
+    try {
+        const serverOptions = host === undefined ? { port } : { host, port };
+        console.log("INFO: running with serverOptions: ", serverOptions);
+        
+        const server = new OPCUAServer(serverOptions);
+        await server.start();
+
+        const isServerListening = await checkServer(host, port);
+        isServerListening.should.be.true;
+
+        await server.shutdown();
+        server.dispose();
+    } catch (err) {
+        should.fail(err, undefined, `Error in server startup/shutdown: ${err.message}`);
+    }
+}
+
+describe("OPCUAServer - issue#1303", () => {
+    it("should start a net.Server on loopback address", async () => {
+        const port = await findAvailablePort();
+        const host = "127.0.0.1";
+        await testServerStartupAndShutdown(host, port);
```

Comment:
We may want to verify here that the server is reachable on the 127.0.0.1 loopback interface and not on the ip.